<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DriverEdit</class>
 <widget class="QDialog" name="DriverEdit">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>320</width>
    <height>183</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Drivers</string>
  </property>
  <widget class="QDialogButtonBox" name="buttonBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>150</y>
     <width>301</width>
     <height>32</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="standardButtons">
    <set>QDialogButtonBox::Close</set>
   </property>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>9</y>
     <width>291</width>
     <height>141</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="2" column="3">
     <spacer name="horizontalSpacer_4">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="3" colspan="2">
     <spacer name="horizontalSpacer_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="lbl_min">
      <property name="text">
       <string>TextLabel</string>
      </property>
      <property name="buddy">
       <cstring>doubleSpinBox</cstring>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <spacer name="horizontalSpacer_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="2">
     <widget class="QComboBox" name="comboBox"/>
    </item>
    <item row="0" column="0" colspan="2">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="2" column="4">
     <widget class="QLabel" name="lbl_max">
      <property name="text">
       <string>TextLabel</string>
      </property>
      <property name="buddy">
       <cstring>doubleSpinBox</cstring>
      </property>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QDoubleSpinBox" name="doubleSpinBox">
      <property name="minimum">
       <double>99.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.030000000000000</double>
      </property>
     </widget>
    </item>
    <item row="1" column="0" colspan="5">
     <widget class="QSlider" name="slider">
      <property name="styleSheet">
       <string notr="true"> QSlider::groove:vertical {
     background: red;
     position: absolute; /* absolutely position 4px from the left and right of the widget. setting margins on the widget should work too... */
     left: 4px; right: 4px;
 }

 QSlider::handle:vertical {
     height: 10px;
     background: green;
     margin: 0 -4px; /* expand outside the groove */
 }

 QSlider::add-page:vertical {
     background: white;
 }

 QSlider::sub-page:vertical {
     background: pink;
 }</string>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>1000</number>
      </property>
      <property name="singleStep">
       <number>1</number>
      </property>
      <property name="pageStep">
       <number>0</number>
      </property>
      <property name="value">
       <number>500</number>
      </property>
      <property name="sliderPosition">
       <number>500</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="invertedAppearance">
       <bool>false</bool>
      </property>
      <property name="tickPosition">
       <enum>QSlider::NoTicks</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>doubleSpinBox</tabstop>
  <tabstop>buttonBox</tabstop>
  <tabstop>comboBox</tabstop>
  <tabstop>slider</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>comboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>DriverEdit</receiver>
   <slot>NewSelection(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>159</x>
     <y>30</y>
    </hint>
    <hint type="destinationlabel">
     <x>303</x>
     <y>19</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>DriverEdit</receiver>
   <slot>HandleSlider(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>50</x>
     <y>82</y>
    </hint>
    <hint type="destinationlabel">
     <x>32</x>
     <y>28</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>DriverEdit</receiver>
   <slot>HandleSpinBox(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>253</x>
     <y>99</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DriverEdit</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DriverEdit</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>SetDriverValue()</slot>
  <slot>GetDriverValues(int)</slot>
  <slot>RequestDriverValue()</slot>
  <slot>ResetSlider()</slot>
  <slot>HandleSlider(int)</slot>
  <slot>HandleSpinBox(double)</slot>
  <slot>RequestDriverParameters()</slot>
  <slot>NewSelection(int)</slot>
 </slots>
</ui>
